R/api-functions.R

```diff
@@ -54,50 +66,61 @@ reserve_edi_id <- function(user_id, password, environment = "production") {
 #' \dontrun{evaluate_edi_package(user_id = "samuelwright", 
 #'                               eml_file_path = "data/edi20.1.xml")}
 #' @export   
-evaluate_edi_package <- function(user_id, password, eml_file_path, environment = "staging") {
-  # Select package environment 
+evaluate_edi_package <-
+  function(user_id,
+           password,
+           eml_file_path,
+           environment = "staging",
+           max_timeout = 60) {
+    # Select package environment 
   base_url <- dplyr::case_when(environment == "staging" ~ "https://pasta-s.lternet.edu/package/",
                                environment == "development" ~ "https://pasta-d.lternet.edu/package/",
                                environment == "production" ~ "https://pasta.lternet.edu/package/")
   
   # post package to EDI for evaluation 
   response <- httr::POST(
     url = paste0(base_url, "evaluate/eml"),
-    config = httr::authenticate(paste('uid=', user_id, ",o=EDI", ',dc=edirepository,dc=org'), password),
-    body = httr::upload_file(eml_file_path)
+    config = httr::authenticate(paste0('uid=', user_id, ",o=EDI", ',dc=edirepository,dc=org'), password),
+    body = httr::upload_file(eml_file_path),
+    timeout = httr::timeout(max_timeout)
   )
   if (response$status_code == "202") {
     # pull transaction id from response content 
     transaction_id <- httr::content(response, as = 'text', encoding = 'UTF-8')
-    iter <- 0
-    max_iter <- 5
-    while(TRUE){ # Loop through a few times to give EDI time to evaluate package 
-      Sys.sleep(2) 
+    # iter <- 0
+    # max_iter <- 5
+    # while(TRUE){ # Loop through a few times to give EDI time to evaluate package 
+    #   Sys.sleep(3) 
       # use transaction id to read evaluation report
-      response<- httr::GET(
-        url = paste0(base_url, "evaluate/report/eml/", transaction_id),
-        config = httr::authenticate(paste('uid=', user_id, ",o=EDI", ',dc=edirepository,dc=org'), password)
-      )
-      iter <- iter + 1
-      if (response$status_code == "200") {
-        # use generate_report_df() function defined above to parse transaction_response 
-        # content into a report_df table 
-        report_df <- generate_report_df(response)
-        print("Please check for errors in the report dataframe")
-        return(report_df)
-        break
-      }
-      if (max_iter > iter) {
-        print("Request timed out, check that you inputs are all valid and try again")
-        break 
-      }
+    response <- httr::GET(
+      url = paste0(base_url, "evaluate/report/eml/", transaction_id),
+      config = httr::authenticate(
+        paste0('uid=', user_id, ",o=EDI", ',dc=edirepository,dc=org'),
+        password,
```

Comment:
```suggestion
        password
```

There is an extra comma that was throwing an error